src/MessageTrait.php

```diff
@@ -66,7 +66,7 @@ public function getProtocolVersion()
      * new protocol version.
      *
      * @param string $version HTTP protocol version
-     * @return self
+     * @return static
```

Comment:
> Considering that `MessageInterface` is not expected to be directly implemented

It's an interface. It will be implemented, and nobody says that it will necessarily be a `RequestInterface` or a `ResponseInterface` :-)